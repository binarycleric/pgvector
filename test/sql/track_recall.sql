SET pgvector.track_recall = on;
SET pgvector.recall_sample_rate = 1;
SET pgvector.recall_max_scan_tuples = -1;

SELECT pg_vector_recall_stats();
SELECT pg_vector_recall_summary();

CREATE TABLE t (val vector(3));
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), (NULL);
CREATE INDEX t_idx ON t USING hnsw (val vector_l2_ops);

SELECT pg_vector_recall_get('t_idx'::regclass::oid);
SELECT pg_vector_recall_reset('t_idx'::regclass::oid);

DROP TABLE t;

SET pgvector.track_recall = off;
SET pgvector.recall_sample_rate = 100;
SET pgvector.recall_max_scan_tuples = 10000;
