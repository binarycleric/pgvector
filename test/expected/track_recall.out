SET pgvector.track_recall = on;
SET pgvector.recall_sample_rate = 1;
SET pgvector.recall_max_scan_tuples = -1;
SELECT pg_vector_recall_stats();
 pg_vector_recall_stats 
------------------------
(0 rows)

SELECT pg_vector_recall_summary();
 pg_vector_recall_summary 
--------------------------
(0 rows)

CREATE TABLE t (val vector(3));
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), (NULL);
CREATE INDEX t_idx ON t USING hnsw (val vector_l2_ops);
SELECT pg_vector_recall_get('t_idx'::regclass::oid);
 pg_vector_recall_get 
----------------------
                     
(1 row)

SELECT pg_vector_recall_reset('t_idx'::regclass::oid);
 pg_vector_recall_reset 
------------------------
 
(1 row)

DROP TABLE t;
SET pgvector.track_recall = off;
SET pgvector.recall_sample_rate = 100;
SET pgvector.recall_max_scan_tuples = 10000;
